src/pages/HomePage.js

```diff
@@ -1,34 +1,56 @@
 import React, { useState, useRef, useEffect } from "react";
-
+import { makeStyles } from "@material-ui/core";
 import MainView from "../components/HomePage/MainView";
-import SearchResultsWidget from "../components/HomePage/SearchResultsArea/SearchResultsWidget/SearchResultsWidget";
-import ProductDescription from "../components/HomePage/ProductPlacementArea/ProductDescription";
 import ApplicationMessagesNotifier from "../components/ApplicationMessages";
 import { smoothScrollToRef } from "../utils";
+import { useDesktop } from "../utils/media-queries";
+import ProductDescription from "../components/HomePage/ProductPlacementArea/ProductDescription";
+import SearchResultsWidget from "../components/HomePage/SearchResultsArea/SearchResultsWidget/SearchResultsWidget";
+
+const useStyles = makeStyles(() => ({
+    searchResults: {
+        scrollSnapAlign: "start",
+        scrollMarginBottom: "100vh", // don't apply snap scroll to the bottom of the search results, so that it's possible to see the footer
+    },
+}));
 
 export const HomePage = () => {
 
     const productDescriptionRef = useRef(null);
     const searchResultsRef = useRef(null);
     const [showSearchResults, setShowSearchResults] = useState(false);
+    const productDescriptionScrollBlock = useDesktop() ? "end" : "center";
 
     // this will not trigger the scroll on subsequent submits, because the dependencies won't change after the first call
     useEffect(() => {
         if (showSearchResults && searchResultsRef) smoothScrollToRef(searchResultsRef);
+
+        // check if the current browser is Firefox, because scroll scroll-snap does not work correctly in Firefox
+        const isFirefox = typeof InstallTrigger !== "undefined";
+
+        // It's mandatory to add the scroll-snap-type style property to the html tag in order to use snap-scroll
+        if (!isFirefox) document.getElementsByTagName("html")[0].style.scrollSnapType = "y proximity";
```

Comment:
I think the solution is to add the `scroll-snap-align` property to the other components in the page as well (since there are none currently, firefox is not able to know what to do when scrolling since it doesn't know where to snap next) I tried it on the netlify deployment and it seems to have worked. You need to add it to all elements so that is scrolls all the way up (if you only add it to product description, it will only scroll up to that div, for example)

Reply:
https://developers.google.com/web/updates/2018/07/css-scroll-snap

I suggest playing with scroll-padding/margin to improve the experience as well (control the snaps threshold)

Reply:
I decided to only apply the scroll-snap behavior when the search results are visible. When they are, there are 3 elements with the `scroll-snap-align` property: the top search, the product description, and the search results. This behavior works well in firefox and chrome (probably the same for the other browsers), as well as in mobile.